.github/workflows/release.yaml

```diff
@@ -0,0 +1,72 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  check:
+    name: preliminary-checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Ensure version is properly set
+        run: |
+          test "$(npm -s run get-version)" = "${{ inputs.tag }}"
```

Comment:
as for version check, there are actually more:
* checking the Tag matches the package
* checking the Tag matches the branch
* checking the Tag has not already been released
* checking the Tag is exactly the "next one" (since previous release)
* (in premerge) checking the package version is accurate, to avoid introducing a problem

this however is probably more related to the version management, or even a separate issue altogether, and should thus probably be addressed separately.